tests/components/dexcom/test_init.py

```diff
@@ -0,0 +1,61 @@
+"""Test the Dexcom config flow."""
+from pydexcom import AccountError, SessionError
+
+from homeassistant.components.dexcom.const import DOMAIN
+from homeassistant.config_entries import ENTRY_STATE_LOADED, ENTRY_STATE_NOT_LOADED
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+from tests.components.dexcom import CONFIG, init_integration
+
+
+async def test_setup_entry_account_error(hass):
+    """Test entry setup failed due to account error."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        title="test_username",
+        unique_id="test_username",
+        data=CONFIG,
+        options=None,
+    )
+    with patch(
+        "homeassistant.components.dexcom.Dexcom", side_effect=AccountError,
+    ):
+        entry.add_to_hass(hass)
+        result = await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert result is False
+
+
+async def test_setup_entry_session_error(hass):
+    """Test entry setup failed due to session error."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        title="test_username",
+        unique_id="test_username",
+        data=CONFIG,
+        options=None,
+    )
+    with patch(
+        "homeassistant.components.dexcom.Dexcom", side_effect=SessionError,
+    ):
+        entry.add_to_hass(hass)
+        result = await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert result is False
+
+
+async def test_unload_entry(hass):
+    """Test successful unload of entry."""
+    entry = await init_integration(hass)
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)